src/peer_digest.cc

```diff
@@ -194,12 +195,12 @@ peerDigestCheck(void *data)
 
     pd->times.next_check = 0;   /* unknown */
 
-    if (!pd->peer.valid()) {
+    if (pd->peer.set() && !pd->peer.valid()) {
         peerDigestNotePeerGone(pd);
         return;
     }
 
-    debugs(72, 3, "cache_peer " << *pd->peer);
+    debugs(72, 3, "cache_peer " << RawPointer(pd->peer.raw()).orNil());
```

Comment:
If possible, please simplify (and make it safer if this code is copied or moved in the future):

```suggestion
    debugs(72, 3, "cache_peer " << RawPointer(pd->peer).orNil());
```